Adds cloud and database preview pages (#4998)
gravitational/teleport: docs/5.0/preview/cloud.md

---
title: Teleport Cloud Preview
description: Teleport hosted and managed by Teleport team.
---

# Teleport Cloud Preview

Teleport cloud is a managed an hosted service of Teleport.
Request access to preview [here](https://goteleport.com/get-started/).

## Status

Teleport cloud is currently only recommended for environments that do not require
meeting strict compliance standards or SLA uptime guarantees.

> We believe that Teleport cloud environment is secure, however we scrutinizing

Reviewer: `however we scrutinizing` reads a bit weird to me. Maybe something more like `however, we are in the process of scrutinizing and executing...`

and executing on our security roadmap and working with independent security
auditors to identify any gaps.

## Roadmap

For the first half of 2021, we will be working with auditors to realize the
threat modeling and security controls.

Once we realize, test and audit the security model we will start working towards meeting
compliance programs such as FedRamp.

## Billing

Usage will be collected, but billing will not be set up until February.
Preview version will be free of charge to preview customers until
we commit to SLA in February.

> ## Security

Reviewer: I still don't believe this section really outlines the security model, and would be better not to include until we can go through in sufficient detail.

Author: ok move to FAQ.

Each customer Teleport is running as separate deployment
in a multi-tenant Kubernetes cluster on the AWS infrastructure.

Each deployment is using at rest encryption for customer
data including session recordings and user records.

Password hashes are generated using
[Golang's bcrypt](https://pkg.go.dev/golang.org/x/crypto/bcrypt).

Teleport uses [letsencrypt.org](https://letsencrypt.org/) to issue
certificates for every customer.

## Managed Teleport Settings

SSH sessions are recorded [on nodes](/architecture/nodes/#session-recording).

Teleport does not terminate SSH sessions when using OpenSSH and `tsh` sessions.

It terminates Application, Database and Kubernetes session on the proxy.

## Data retention

The session records are stored in S3 using at rest encryption and
are not expired and retention policies are not set.

> This is subject to change after GA release.

Reviewer: I would probably just say this has yet to be determined and is subject to change, instead of saying what the current temporary state is.

Author: Done. Moved questions to FAQ.

Customer data including session events is backed up using DynamodDB
point in time backup system with recovery option up to 35 days
and is not configurable.

## High availability

Clusters are deployed in a single AWS region in 2 availability zones.
AWS guarantees [99.99%](https://aws.amazon.com/compute/sla/) of monthly uptime percentage.

Teleport cloud preview version does not commit to any SLA. It sets availability
objective to 99.7% of monthly uptime percentage or maximum of two hours of downtime per month.

> We plan to get to 99.99% uptime SLA in February.

Reviewer: I don't see how this is possible.

Author: ok, removed specific date, but we'd have to commit to some SLA in Feb. We can discuss offline

## Roadmap

For the first half of 2021, we will be working with auditors to realize the
threat modeling and security controls.

Once we realize, test and audit the security model we will start working towards meeting
compliance programs such as FedRamp.

## Billing

Usage will be collected, but billing will not be set up until February.
Preview version will be free of charge to preview customers until
we commit to SLA in February.

## Managed Teleport Settings

SSH sessions are recorded [on nodes](../architecture/nodes.md#session-recording).

Teleport does not terminate SSH sessions when using OpenSSH and `tsh` sessions.

It terminates Application, Database and Kubernetes session on the proxy.

## Data retention

The session records are stored in S3 using at rest encryption.
We have yet to define specific retention policies.

Customer data including session events is backed up using DynamodDB
point in time backup system with recovery option up to 35 days
and is not configurable.

## High availability

Clusters are deployed in a single AWS region in 2 availability zones.
AWS guarantees [99.99%](https://aws.amazon.com/compute/sla/) of monthly uptime percentage.

> Teleport cloud preview version does not commit to any SLA. It sets availability

Reviewer 2: ```suggestion
Teleport Cloud preview version does not commit to any SLA. It sets availability
```

> objective to 99.7% of monthly uptime percentage or maximum of two hours of downtime per month.

Reviewer 2: ```suggestion
objective to 99.7% of monthly uptime percentage (a maximum of two hours of downtime per month).
```

We plan to get to 99.99% uptime SLA.

# FAQ

### Can I use Teleport Cloud Preview in production?

It should be OK to use Cloud in non mission critical systems that do not
require 24/7 access or when fallback option is available.

> ### Are you using AWS Managed Keys, or CMKs via KMS?

Reviewer 2: ```suggestion
### Are you using AWS-managed encryption keys, or CMKs via KMS?
```

We use AWS-managed keys. There is currently no option to provide your own key.

### Is this Teleport's S3 bucket, or my bucket based on my AWS credentials?

It's a Teleport-managed S3 bucket with AWS-managed keys.
There is no way to provide your own bucket.

### How do I add nodes to Teleport Cloud?

You can connect servers, kubernetes clusters, databases and applications
using [reverse tunnels](../admin-guide.md#adding-a-node-located-behind-nat).

There is no need to open any ports on your infrastructure for inbound traffic.

### How can I access the tctl admin tool?

> You can login using tsh with your credentials and then use `tctl`

Reviewer 2: ```suggestion
We have made changes to allow you to log into your cluster using `tsh`, then use `tctl` remotely:
```

```bash
$ tsh login --proxy=myinstance.teleport.sh:443

> # tctl will recognize SSO credentials starting version 5.0

Reviewer 2: ```suggestion
# tctl (Enterprise edition) can use tsh credentials as of version 5.0
```

$ tctl status
```

### Are dynamic node tokens available?

> After [connecting](#how-can-i-access-the-tctl-admin-tool) `tctl` to the cloud, users can generate [dynamic tokens](admin-guide/#short-lived-dynamic-tokens)

Reviewer 2: ```suggestion
After [connecting](#how-can-i-access-the-tctl-admin-tool) `tctl` to Teleport Cloud, users can generate [dynamic tokens](../admin-guide.md#short-lived-dynamic-tokens)
```

> using `tctl` with cloud instance:

Reviewer 2: ```suggestion
using `tctl`:
```

```bash
tctl nodes add --ttl=5m --roles=node,proxy --token=$(uuid)
```

### When will a security audit be available?

> The cloud audit starts in January and will publish it in Q1 2021 after fixing

Reviewer 2: ```suggestion
The Teleport Cloud security audit starts in January. We will publish it in Q1 2021 after fixing
```

any security issues found.

### What does Teleport Cloud run in?

> It's using [gravity](https://github.com/gravitational/gravity) cluster on AWS.

Reviewer 2: ```suggestion
Teleport Cloud is deployed using a [Gravity](https://github.com/gravitational/gravity) cluster on AWS.
```

### Will Teleport be automatically upgraded with each release?

> We will be upgrading preview version automatically.

Reviewer 2: ```suggestion
We will be upgrading the preview version of Teleport Cloud automatically.
```

> We will add an option to trigger upgrades manually and automatically.

Reviewer 2: ```suggestion
We will add an option to trigger upgrades manually and automatically (within a maintenance window).
```

### Does your SOCII report include Teleport Cloud?

Our current SOCII Type2 report covers our organization, Teleport and Gravity OSS and Enterprise versions.
We are pursuing SOC2 for cloud and expect to receive our report at the end of Q2 2021
after our 6 month audit period is completed.

### Can a customer deploy multiple clusters in Teleport Cloud?

> Not at this time

Reviewer 2: ```suggestion
Not at this time.
```

> ### Is FIPS an option or it's always FIPS?

Reviewer 2: ```suggestion
### Is FIPS mode an option?
```

> FIPS is not currently an option for cloud version.

Reviewer 2: ```suggestion
FIPS is not currently an option for Teleport Cloud clusters.
```

### How do you store passwords?

Password hashes are generated using
[Golang's bcrypt](https://pkg.go.dev/golang.org/x/crypto/bcrypt).

### How does teleport manage web certificates? Can I upload my own?

Teleport uses [letsencrypt.org](https://letsencrypt.org/) to issue

> certificates for every customer. It is not possible to upload custom

Reviewer 2: ```suggestion
certificates for every customer. It is not possible to upload a custom
```

> certificate or use custom domain name.

Reviewer 2: ```suggestion
certificate or use a custom domain name.
```

### Do you encrypt data at rest?

> Each deployment is using at rest encryption using AWS DynamoDB and S3 at rest encryption

Reviewer 2: ```suggestion
Each deployment is using at-rest encryption using AWS DynamoDB and S3 at-rest encryption
```

---
title: Teleport Cloud Preview
description: Teleport hosted and managed by the Teleport team.
---

# Teleport Cloud Preview

Teleport Cloud is a hosted, managed Teleport cluster. 
Currently Teleport Cloud is in preview status and we are maintaining a 
wait-list of interested parties wanting access to the service. 
To request access age get on the waiting list [here](https://goteleport.com/get-started/).

## Status

Teleport Cloud is currently only recommended for environments that do not require
strict compliance standards or SLA uptime guarantees.

We have designed Teleport Cloud environment to be secure; however we are still in the process of
scrutinizing and executing on our security roadmap and working with independent security
auditors to identify any gaps. Only once this is complete will we be able to to verify 

> that it is ready for strict compliance use-cases.

Reviewer 3: ```
Only once this is complete will we be able to to verify 
that it is ready for strict compliance use-cases.
```

This ^ sentence read a little choppy. 

```
Once this is complete, the team will evaluate whether the 
platform is ready for strict compliance use-cases. 
```

This retention period is not configurable.

## High availability

Clusters are deployed in a single AWS region in 2 availability zones.
AWS guarantees [99.99%](https://aws.amazon.com/compute/sla/) of monthly uptime percentage.

Teleport Cloud preview version does not commit to any SLA. It sets availability
objective to 99.7% of monthly uptime percentage (a maximum of two hours of downtime per month).

We plan to get to 99.99% uptime SLA.

# FAQ

### Can I use Teleport Cloud Preview in production?

We expect Teleport Cloud to be OK to use in non mission critical systems
that do not require 24/7 access or a guaranteed SLA. We recommend a 
fallback option be make available. 

### Are you using AWS-managed encryption keys, or CMKs via KMS?

We use AWS-managed keys. Currently there is no option to provide your own key.

### Is this Teleport's S3 bucket, or my bucket based on my AWS credentials?

It's a Teleport-managed S3 bucket with AWS-managed keys.
Currently there is no way to provide your own bucket.

### How do I add nodes to Teleport Cloud?

You can connect servers, kubernetes clusters, databases and applications

> using [reverse tunnels](../admin-guide.md#adding-a-node-located-behind-nat).

Reviewer 4: Note to self: The 'Reverse Tunnels' content could do a with a refresh

> There is no need to open any ports on your infrastructure for inbound traffic.

Reviewer 4: What's the outbound requirements for strict network requirements, should we also advise any IPs/FQDN to whitelist?

Author: We don't promise to keep any IPs, I don't think AWS promises anything either

Reviewer 2: They can whitelist [these](https://ip-ranges.amazonaws.com/ip-ranges.json) if they like :)

### When will a security audit be available?

The Teleport Cloud security audit starts in January. 
We will work on fixing any security issues that are identified and 
publish it in Q1 2021.

> ### What does Teleport Cloud run in?

Reviewer 5: This didn't make sense to me. But if it does to you that is fine. I would think that we use Gravity to deploy and it runs in K8s?
